cpp/open3d/geometry/PointCloud.cpp

```diff
@@ -514,6 +518,10 @@ PointCloud::RemoveRadiusOutliers(size_t nb_points, double search_radius) const {
         size_t nb_neighbors = kdtree.SearchRadius(points_[i], search_radius,
                                                   tmp_indices, dist);
         mask[i] = (nb_neighbors > nb_points);
+        if (print_progress && (i + 1) > int(progress_bar.GetCurrentCount())) {
+            // In parallel loop, progess bar maybe up and down...
```

Comment:
Let's do it properly. We'll need to handle concurrency inside a parallel for loop. See the discussions in https://stackoverflow.com/q/28050669/1255535.

Here are my suggestions:
1. Wait till #4254 is merged. This will move the progress bar to a standalone file so it is easier to work with.
2. I would sub-class `ProgressBar` to be `OMPProgressBar` to make our intention explicit.
3. In `ParallelProgressBar::operator++()`, 1) check if you're inside an OpenMP for loop, 2) check how many threads are used, 3) update your percentage only when thread_id == 0 (taking the idea from https://stackoverflow.com/a/44555438/1255535).
4. Add a unit test (`cpp/tests/utility/ProgressBar.cpp`) for a simple for loop and the new OMP progress bar.

Reply:
thanks for your reply, I will fix this later.
May I ask if your third suggestion is to express `OMPProgressBar::operator++()`?

Reply:
Yes, `OMPProgressBar::operator++()`. I've updated my comment.

Reply:
FYI, @4254 is merged so you may start to refactor the code

Reply:
thanks